public/static/docs/user-guide/setup-google-drive-remote.md

```diff
@@ -0,0 +1,114 @@
+# Setup a Google Drive DVC Remote
+
+Follow this guide to setup Google Drive as your <abbr>DVC project</abbr>'s
+[remote storage](/doc/command-reference/remote).
+
+## Configure a Google Cloud _project_
+
+DVC uses the [Google Drive API](https://developers.google.com/drive) to connect
+to your Google Drive folders. This requires a Google Cloud _project_ that allows
+Drive API connections, and its
+[OAuth](https://developers.google.com/identity/protocols/OAuth2) credentials
+(**client ID** and **client secret**).
+
+1. Sign into the [Google API Console](https://console.developers.google.com).
+
+   > Double check you're using the intended Google account (upper-right corner).
+
+2. Select or
+   [Create](https://cloud.google.com/resource-manager/docs/creating-managing-projects#creating_a_project)
+   a project for DVC remote connections.
+
+3. [Enable the Drive API](https://developers.google.com/drive/api/v2/about-sdk)
+   from the **APIs & Services** **Dashboard** (left sidebar), click on **+
+   ENABLE APIS AND SERVICES**. Find and select the "Google Drive API" in the API
+   Library, and click on the **ENABLE** button.
+
+   ![](/static/img/gdrive-enable-apis-and-services.png)
+
+4. Go back to **APIs & Services** in the left sidebar, and select **OAuth
+   consent screen**. Chose a **User Type** and click **CREATE**. On the next
+   screen, enter an **Application name** e.g. "DVC remote storage", and click
+   the **Save** (scroll to bottom).
+
+5. From the left sidebar, select **Credentials**, and click the **Create
+   credentials** dropdown to select **OAuth client ID**. Chose **Other** and
+   click **Create** to proceed with a default client name.
+
+   ![](/static/img/gdrive-create-credentials.png)
+
+6. The newly generated **client ID** and **client secret** should be shown to
+   you now, and you can always come back to **Credentials** to fetch them.
+
+⚠️ It's safe to share **client ID** and **client secret** among your team. These
+credentials are used to
+[authorize](https://developers.google.com/drive/api/v2/about-auth) DVC to use
+your Google Drive.
```

Comment:
@jorgeorpinel it seems it would be wrong to say that `these credentials are used to authorize DVC to use your Google Drive`. As mentioned above, they are used to give an idea to Google "who" is using Google's API and what kind of restrictions Google should apply for these "who". Previously it was explained as:

These credentialds are used to
[retrieve an authorization token](https://developers.google.com/identity/protocols/OAuth2)
and
[authenticate your application](https://developers.google.com/drive/api/v2/about-auth)
to Google. Credentialds don't give an access to your Google Drive storage.

Reply:
From the link (https://developers.google.com/drive/api/v2/about-auth):

> Every request your application sends to the Drive API must include an authorization token. The token also identifies your application to Google
> ...
> ### Authorizing requests

You authenticate users, but authorize app requests. DVC is the app. This is why I think the term "authorize" is more appropriate here and I also wanted to simplify the longer sentence you had with 2 different links "credentialds are used to retrieve an authorization token and authenticate your application" – both these links already exist in other places in the doc 🙂

Reply:
I, personally, will start panic if someone going to suggest me to share the **client secret** that authorizes DVC to my Google Drive :)

Reply:
Going to update last sentence to the following:
```
These credentials are only used to generate the
[authorization](#authorization) URL DVC will later prompt to visit in order to connect to the
Google Drive.
```